### PR TITLE
Fix for eliding of the ChainDB tracer

### PR DESCRIPTION
**Pull request by jutaro**
The addition of new trace messages made the eliding function inoparable. 'cardano.node.ChainDB:Notice' generates 1000s of log lines per second during sync. This fix restores the functionality

### CODE REVIEW

File: cardano-node/src/Cardano/Tracing/Tracers.hs
Comment (Jasagredo) [inline cardano-node/src/Cardano/Tracing/Tracers.hs:232]: I think Invalid blocks should not be elided. We don't expect finding many of these.

File: cardano-node/src/Cardano/Tracing/Tracers.hs
Comment (Jasagredo) [inline cardano-node/src/Cardano/Tracing/Tracers.hs:233]: Same here, I would leave this one as `False` too.

File: cardano-node/src/Cardano/Tracing/Tracers.hs
Comment (Jasagredo) [inline cardano-node/src/Cardano/Tracing/Tracers.hs:244]: Nitpicking: If we change this format, I would at least omit the second `let`
```suggestion
      let tnow' = fromIntegral tnow
          deltat = tnow' - oldt
```

**Review by Jasagredo — COMMENTED** [review]
Apart from the points I mentioned, I would first ask someone if it is okay to elide pipelining events, probably @nfrisby has some suggestion here.

Otherwise, this change fixes the error for now. I agree with not putting too much work on the legacy framework as it is going to be replaced, so I think this is good enough.

I will approve once we are certain that eliding pipelining events is OK.

File: cardano-node/src/Cardano/Tracing/Tracers.hs
Comment (jutaro) [reply cardano-node/src/Cardano/Tracing/Tracers.hs:232]: Changed it.

File: cardano-node/src/Cardano/Tracing/Tracers.hs
Comment (jutaro) [reply cardano-node/src/Cardano/Tracing/Tracers.hs:233]: Changed it.

Comment (jutaro) [reply cardano-node/src/Cardano/Tracing/Tracers.hs:233]: Changed it.

File: cardano-node/src/Cardano/Tracing/Tracers.hs
Comment (jutaro) [reply cardano-node/src/Cardano/Tracing/Tracers.hs:244]: Changed it

File: cardano-node/src/Cardano/Tracing/Tracers.hs
Comment (nfrisby) [inline cardano-node/src/Cardano/Tracing/Tracers.hs:236]: > The addition of new trace messages made the eliding function inoparable. 'cardano.node.ChainDB:Notice' generates 1000s of log lines per second during sync.

I agree that suggests we should elide pipelining events.

The caveat is that we inspect benchmarking node logs in order to directly confirm pipelining is working as expected, and we won't be able to do that if eliding these events.

Relevant fact: the benchmarking runs (at least the ones we've used so far) don't involve bulk fetch/sync.

Thus, two possible ideas for follow-up:

  * disable pipelining or at least pipelining events during bulk fetch. (It's not entirely obvious how to set this switch, but it seems possible).
  
  * add a config flag we could set during benchmarking runs in order to disable elision of pipelining events

File: cardano-node/src/Cardano/Tracing/Config.hs
Comment (nfrisby) [inline cardano-node/src/Cardano/Tracing/Config.hs:73]: FYI, I don't know the consequences of this. Perhaps someone else should review this line.

File: cardano-node/src/Cardano/Tracing/Tracers.hs
Comment (Jasagredo) [reply cardano-node/src/Cardano/Tracing/Tracers.hs:236]: > add a config flag

> I consider a simple fix

What about defining a small CPP flag that sets this to True for normal running, and False for benchmarking? That should be quite fast and small (although questionable, but it is small enough to be a "temporary fix" if this is going to be removed soon)

File: cardano-node/src/Cardano/Tracing/Config.hs
Comment (Jasagredo) [reply cardano-node/src/Cardano/Tracing/Config.hs:73]: In particular, in the JSON configuration provided in `configuration/cardano/mainnet-config.json` it is written as `"TraceChainDb": true` and without this fix it was not being picked up because it was expecting `"TraceChainDB"` as @jutaro mentions

Comment (Jimbo4350) [reply cardano-node/src/Cardano/Tracing/Config.hs:73]: Nice catch @jutaro

**Review by Jimbo4350 — APPROVED** [review]

**Pull request merged**
